modules/build/src/main/scala/scala/build/internal/markdown/MarkdownCodeBlockUtil.scala

```diff
@@ -0,0 +1,122 @@
+
+/** A util for handling of code blocks in Markdown. */
+object MarkdownCodeBlockUtil {
+  private val allowIndentedFence: Boolean = false
+
+  /** Representation for a code block contained in Markdown
+    *
+    * @param info
+    *   a list of tags tied to a given code block
+    * @param body
+    *   the code block content
+    * @param startLine
+    *   starting line on which the code block was defined
+    * @param endLine
+    *   end line on which the code block was closed
+    */
+  case class MarkdownFence(
+    info: Seq[String],
+    body: String,
+    startLine: Int, // start of fenced body EXCLUDING backticks
+    endLine: Int    // same as above
+  ) {
+    override def toString: String =
+      s"Fence[$info, lines $startLine-$endLine]{${body.replace("\n", "\\n")}}"
+
+    /** @return
+      *   `true` if this snippet should be ignored, `false` otherwise
+      */
+    def shouldIgnore: Boolean = info.head != "scala" || info.contains("ignore")
+
+    /** @return
+      *   `true` if this snippet should have its scope reset, `false` otherwise
+      */
+    def resetScope: Boolean = info.contains("reset")
+
+    /** @return
+      *   `true` if this snippet is a test snippet, `false` otherwise
+      */
+    def isTest: Boolean = info.contains("test")
+
+    /** @return
+      *   `true` if this snippet is a raw snippet, `false` otherwise
+      */
+    def isRaw: Boolean = info.contains("raw")
+  }
+
+  private case class StartedFence(
+    info: String,
+    tickStartLine: Int, // fence start INCLUDING backticks
+    backticks: String,
+    indent: Int
+  )
+
+  /** Closes started code-fence
+    *
+    * @param started
+    *   [[StartedFence]] representing this code-fence's start
+    * @param tickEndLine
+    *   number of the line where closing backticks are
+    * @param lines
+    *   input file sliced into lines
+    * @return
+    *   [[MarkdownFence]] representing whole closed code-fence
+    */
+  private def closeFence(
+    started: StartedFence,
+    tickEndLine: Int,
+    lines: Array[String]
+  ): MarkdownFence = {
+    val start: Int               = started.tickStartLine + 1
+    val bodyLines: Array[String] = lines.slice(start, tickEndLine)
+    MarkdownFence(
+      started.info.split("\\s+").toList, // strip info by whitespaces
+      bodyLines.tail.foldLeft(bodyLines.head)((body, line) => body.:++("\n" + line)),
+      start,          // snippet has to begin in the new line
+      tickEndLine - 1 // ending backticks have to be placed below the snippet
+    )
+  }
+
+  /** Finds all code snippets in given input
+    *
+    * @param md
+    *   Markdown file in a `String` format
+    * @return
+    *   list of all found snippets
+    */
+  def findFences(md: String): Seq[MarkdownFence] = {
+    var startedFenceOpt: Option[StartedFence] = None
+    val fences                                = mutable.ListBuffer.empty[MarkdownFence]
+    val lines: Array[String]                  = md.split("\n\r?")
```

Comment:
It would be safer, if you use `Position.Raw.lineStartIndices(fileContent)` to split content by lines - [here](https://github.com/VirtusLab/scala-cli/blob/2cf64ad2f16f70c720fa878261381dd5c7dadd70/modules/cli/src/main/scala/scala/cli/commands/DependencyUpdate.scala#L102)

Reply:
I ended up rewriting a big chunk of the code in this source file, this should be addressed already